src/host/ut_host/ScreenBufferTests.cpp

```diff
@@ -2502,6 +2506,108 @@ void ScreenBufferTests::SetColorTableThreeDigits()
 
 }
 
+
+void ScreenBufferTests::SetDefaultForegroundColor()
+{
+    // Setting the default foreground color should work
+
+    CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.LockConsole(); // Lock must be taken to swap buffers.
+    auto unlock = wil::scope_exit([&] { gci.UnlockConsole(); });
+
+    SCREEN_INFORMATION& mainBuffer = gci.GetActiveOutputBuffer();
+    VERIFY_IS_FALSE(mainBuffer._IsAltBuffer());
+    WI_SetFlag(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING);
+    VERIFY_IS_TRUE(WI_IsFlagSet(mainBuffer.OutputMode, ENABLE_VIRTUAL_TERMINAL_PROCESSING));
+
+    StateMachine& stateMachine = mainBuffer.GetStateMachine();
+
+    COLORREF originalColor = gci.GetDefaultForegroundColor();
+    COLORREF newColor = gci.GetDefaultForegroundColor();
+    COLORREF testColor = RGB(0x33, 0x66, 0x99);
+    VERIFY_ARE_NOT_EQUAL(originalColor, testColor);
+
+    // Valid hexadecimal notation

```

Comment:
Frequently I make these comments be `Log::Comment(L"Valid Hexadecimal Notation")` instead so you can see the sub-tests in the output logs instead of mentally matching up the 4th verify with a particular subtest. Something to consider.

Reply:
I've literally never written unit tests in C++ and it took me 15 minutes to figure out how to even run these (and I still don't know how to get them to run in Visual Studio), that tip is appreciated.